src/compas_notebook/conversions/geometry.py

```diff
@@ -93,6 +135,245 @@ def polyline_to_threejs(polyline: Polyline) -> three.BufferGeometry:
+    return geometry
+
+
+def ellipse_to_threejs(ellipse: Ellipse, max_angle: float = 5.0) -> three.BufferGeometry:
+    """Convert a COMPAS ellipse to PyThreeJS.
+
+    Parameters
+    ----------
+    ellipse : :class:`compas.geometry.Ellipse`
+        The ellipse to convert.
+    max_angle : float, optional
+        Maximum angle in degrees between segments.
+
+    Returns
+    -------
+    :class:`three.BufferGeometry`
+
+    """
+    import math
+
+    n = max(8, int(math.ceil(360.0 / max_angle)))
+    polyline = ellipse.to_polyline(n=n)
+    vertices = numpy.array(polyline.points, dtype=numpy.float32)
+    geometry = three.BufferGeometry(attributes={"position": three.BufferAttribute(vertices, normalized=False)})
+    return geometry
+
+
+def vector_to_threejs(vector: Vector, scale: float = 1.0) -> list[three.Object3D]:
+    """Convert a COMPAS vector to PyThreeJS as arrow.
+
+    Parameters
+    ----------
+    vector : :class:`compas.geometry.Vector`
+        The vector to convert.
+    scale : float, optional
+        Scale factor for vector length.
+
+    Returns
+    -------
+    list[three.Object3D]
+        Line and cone representing the arrow.
+
+    """
+    # Line from origin to vector * scale
+    line = Line([0, 0, 0], vector * scale)
+    line_geom = line_to_threejs(line)
+    line_obj = three.Line(line_geom, three.LineBasicMaterial(color="blue"))
+
+    # Cone head at tip (10% of length, positioned at end)
+    length = vector.length * scale
+    cone_height = length * 0.1
+    cone_radius = cone_height * 0.3
+
+    # Position cone at vector tip
+    cone_geom = three.CylinderGeometry(
+        radiusTop=0,
+        radiusBottom=cone_radius,
+        height=cone_height,
+        radialSegments=8,
+    )
+    cone_obj = three.Mesh(cone_geom, three.MeshBasicMaterial(color="blue"))
+
+    # Compute cone position and rotation
+    # Cone should point in vector direction
+    tip = vector * scale
+    cone_obj.position = (tip.x, tip.y, tip.z)
+
+    # Align cone with vector direction
+    # Three.js cylinder default is Y-up, need to rotate to align with vector
+    normalized = vector.unitized()
+    cone_obj.quaternion = _vector_to_quaternion(normalized)
+
+    return [line_obj, cone_obj]
+
+
+def _vector_to_quaternion(vector):
+    """Helper to compute quaternion for aligning Y-axis with vector."""
+    import math
+
+    # Default direction is Y-up [0, 1, 0]
+    y_axis = Vector(0, 1, 0)
+
+    # Compute rotation axis (cross product)
+    axis = y_axis.cross(vector)
+
+    # Compute rotation angle
+    angle = math.acos(max(-1, min(1, y_axis.dot(vector))))
+
+    if axis.length < 1e-10:
+        # Vector is parallel or anti-parallel to Y-axis
+        if vector.y > 0:
+            return (0, 0, 0, 1)  # No rotation
+        else:
+            return (0, 0, 1, 0)  # 180 degree rotation around Z
+
+    # Normalize axis
+    axis = axis.unitized()
```

Comment:
unless you really want to make an independent copy, this is the same as `axis.unitize()`